orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/gce/SetStatefulDiskTask.java

```diff
@@ -0,0 +1,80 @@
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.clouddriver.KatoService;
+import com.netflix.spinnaker.orca.clouddriver.model.TaskId;
+import com.netflix.spinnaker.orca.clouddriver.pipeline.providers.gce.SetStatefulDiskStage.StageData;
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.support.TargetServerGroup;
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.support.TargetServerGroupResolver;
+import com.netflix.spinnaker.orca.clouddriver.tasks.AbstractCloudProviderAwareTask;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import java.util.List;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import rx.Observable;
+
+@Component
+public class SetStatefulDiskTask extends AbstractCloudProviderAwareTask {
+
+  private static final String KATO_OP_NAME = "setStatefulDisk";
+
+  private final KatoService katoService;
+  private final TargetServerGroupResolver resolver;
+
+  @Autowired
+  public SetStatefulDiskTask(KatoService katoService, TargetServerGroupResolver resolver) {
+    this.katoService = katoService;
+    this.resolver = resolver;
+  }
+
+  @Override
+  public TaskResult execute(Stage stage) {
+
+    StageData data = stage.mapTo(StageData.class);
+
+    List<TargetServerGroup> resolvedServerGroups = resolver.resolve(stage);
+    checkArgument(
+        resolvedServerGroups.size() > 0,
+        "Could not find a server group named %s for %s in %s",
+        data.serverGroupName,
+        data.accountName,
+        data.region);
+    checkState(
+        resolvedServerGroups.size() == 1,
+        "Found multiple server groups named %s for %s in %s",
+        data.serverGroupName,
+        data.accountName,
+        data.region);
+    TargetServerGroup serverGroup = resolvedServerGroups.get(0);
+
+    ImmutableMap<String, String> opData =
+        ImmutableMap.<String, String>builder()
+            .put("credentials", getCredentials(stage))
+            .put("serverGroupName", serverGroup.getName())
+            .put("region", data.getRegion())
+            .put("deviceName", data.deviceName)
+            .build();
+
+    Map<String, Map> operation = ImmutableMap.of(KATO_OP_NAME, opData);
+    Observable<TaskId> observable =
+        katoService.requestOperations(getCloudProvider(stage), ImmutableList.of(operation));
+    observable.toBlocking().first();
+
+    ImmutableMap<String, ImmutableList<String>> modifiedServerGroups =
+        ImmutableMap.of(data.getRegion(), ImmutableList.of(serverGroup.getName()));
+    ImmutableMap<String, Object> context =
+        ImmutableMap.<String, Object>builder()
```

Comment:
Not familiar with the `.<type>` notation, is this just parameterizing the builder?

Reply:
Yeah, since there's an intermediate object (the Builder), the compiler can't automatically figure out the type for it... so you have to tell it.